packages/json-api/src/Registry.ts

```diff
@@ -0,0 +1,92 @@
+import { JSONValue, JSONAPIDocument, JSONAPIResource } from "./jsonApiTypes";
+import {
+  IncludesMap,
+  MapObject,
+  Registry,
+  Resource,
+  ResourceParseOptions,
+  ResourceSpec
+} from "./types";
+import { identity, kebabCaseDeep, camelCaseDeep, TransformFunc } from "./utils";
+import ResourceImpl from "./Resource";
+
+export type RegistryCreateOptions = {
+  keyTransform?: "kebab";
+};
+
+export default class RegistryImpl implements Registry {
+  private readonly options?: RegistryCreateOptions;
+
+  private readonly resources: MapObject<Resource>;
+
+  public readonly keyTransformFunc: TransformFunc;
+
+  public readonly keyParseFunc: TransformFunc;
+
+  constructor(options?: RegistryCreateOptions) {
+    this.options = options;
+    this.resources = {};
+
+    this.keyTransformFunc =
+      this.options?.keyTransform === "kebab" ? kebabCaseDeep : identity;
+    this.keyParseFunc =
+      this.options?.keyTransform === "kebab" ? camelCaseDeep : identity;
+  }
+
+  define(type: string, spec: ResourceSpec): Resource {
+    const resource = new ResourceImpl(this, type, spec);
+    if (type in this.resources) {
+      throw new Error(`Resource ${type} is already defined`);
+    }
+    this.resources[type] = resource;
+    return resource;
+  }
+
+  find(type: string): Resource | null {
+    return this.resources[type];
+  }
```

Comment:
Add validation for find function

```suggestion
  find(type: string): Resource | null {
    if (!this.resources[type]) {
      throw new Error(`Cannot find resource ${type}`);
    }
    return this.resources[type];
  }
```

Reply:
This actually changes library API and makes using it in our context more difficult.

Reply:
Also, unknown resource type it is an expected situation for find() (in contrast to format()).

Reply:
Fair point